### PR TITLE
refactor(pubsub): s/SaveMessage/AddMessage

### PR DESCRIPTION
**Pull request by alevenberg**
This is a no-op

https://github.com/googleapis/google-cloud-cpp/issues/13171

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13179)

### CODE REVIEW

File: google/cloud/pubsub/internal/message_batch.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ class MessageBatch {
 
   // Saves the message from the Publish call. Invoked in
   // `BatchingPublisherConnection::Publish(...)`.
-  virtual void SaveMessage(pubsub::Message m) = 0;
+  virtual void AddMessage(pubsub::Message m) = 0;
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/message_batch.h:42]: While you are changing the signature, this should also be a `const&`.

**Review by dbolduc — COMMENTED** [review]
Why is this PR useful?

File: google/cloud/pubsub/internal/message_batch.h (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/message_batch.h:42]: Already was working on it when I saw this comment. Nice.

**Comment by alevenberg** [comment]
> Why is this PR useful?

Debatable if it is useful. I'm going to move the files from the MessageBatch -> BatchSink and if these changes are in the same PR I thought it would be a messier diff and more annoying to review. I guess in retrospect I could have made this a commit. Oh well 🤷

**Review by dbolduc — APPROVED** [review]

**Pull request merged**